applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c

```diff
@@ -106,6 +119,66 @@ static int next_free_audio_stream_index_get(uint8_t *index)
 	return -ENOSPC;
 }
 
+/**
+ * @brief  Get the presentation delay for all headsets
+ *
+ * @param  mode   The mode of search for the presentation delay
+ * @param  index  The index of the hesdset to test against
+ *
+ * @return The chosen presentation delau in micro seconds or 0 if error
+ */
+static uint32_t find_headset_pres_delay(enum le_audio_pres_search mode, int index)
+{
+	uint32_t pres_dly_us = 0;
+	uint32_t pres_dly_min = headsets[index].pref_qos.pd_min;
+	uint32_t pres_dly_max = headsets[index].pref_qos.pd_max;
+	uint32_t pref_dly_min = headsets[index].pref_qos.pref_pd_min;
+	uint32_t pref_dly_max = headsets[index].pref_qos.pref_pd_max;
+
+	for (int i = 0; i < ARRAY_SIZE(headsets); i++) {
+		if (headsets[i].sink_stream != NULL) {
+			pres_dly_min = MAX(pres_dly_min, headsets[i].pref_qos.pd_min);
+			pres_dly_max = MIN(pres_dly_max, headsets[i].pref_qos.pd_max);
+			pref_dly_min = MAX(pref_dly_min, headsets[i].pref_qos.pref_pd_min);
+			pref_dly_max = MIN(pref_dly_max, headsets[i].pref_qos.pref_pd_max);
+		}
+	}
+
+	switch (mode) {
+	case AUDIO_PRES_DELAY_MODE_MIN:
+		pres_dly_us = pres_dly_min;
+		break;
```

Comment:
Is there any reason for this?
The compiler will generate virtually identical code, just looks a bit messy and if this is updated to further process the delay it will have to change back.

Reply:
It's cleaner, and everything else in this project is done compile time with configurations. The only way to search again with another search mode is either through shell or to create a UI. It also removes a lot of code

Reply:
Update to use "if (IS_ENABLED(.....)" and return from each.